stdlib/public/core/RangeReplaceableCollectionType.swift

```diff
@@ -446,10 +446,10 @@ public func +<
     S : SequenceType
     where S.Generator.Element == C.Generator.Element
 >(lhs: C, rhs: S) -> C {
-  var lhs = lhs
-  // FIXME: what if lhs is a reference type?  This will mutate it.
-  lhs.appendContentsOf(rhs)
-  return lhs
+  var result = lhs is AnyObject ? C(lhs) : lhs
```

Comment:
It would be easy to implement, but not required to implement.  The default implementation does not do it (and can't, given the current language features).